server/src/definition/MacroDefinitionProvider.ts

```diff
@@ -31,17 +31,22 @@ export class MacroDefinitionProvider implements DbtNodeDefinitionProvider {
       }
 
       const [, macro] = macroMatch;
-      const macroSearchIds = macro.includes('.')
-        ? [`macro.${macro}`]
-        : [`macro.${MacroDefinitionProvider.DBT_PACKAGE}.${macro}`, `macro.${packageName}.${macro}`];
-      const foundMacro = this.dbtRepository.macros.find(m => macroSearchIds.includes(m.uniqueId));
-      if (foundMacro) {
-        const macroFilePath = path.join(foundMacro.rootPath, foundMacro.originalFilePath);
-        const [definitionRange, selectionRange] = this.getMacroRange(foundMacro.name, macroFilePath);
 
+      const pointIndex = macro.indexOf('.');
+      const packageName = pointIndex !== -1 ? macro.substring(0, pointIndex) : undefined;
+      const macroName = pointIndex !== -1 ? macro.substring(pointIndex + 1) : macro;
+
+      const foundMacros =
+        pointIndex !== -1
+          ? this.dbtRepository.macros.filter(m => m.name === macroName && (packageName === undefined || m.packageName === packageName))
```

Comment:
Please try to get rid of duplicated code.
`this.dbtRepository.macros.filter(m => m.name === macroName).filter(m => pointIndex === -1 || (packageName === undefined || m.packageName === packageName))`?